libvmi/accessors.c

```diff
@@ -380,15 +381,67 @@ status_t vmi_pagetable_lookup_extended(
     info->dtb = dtb;
 
     if(vmi->arch_interface && vmi->arch_interface->v2p) {
-        vmi->arch_interface->v2p(vmi, dtb, vaddr, info);
+        ret = vmi->arch_interface->v2p(vmi, dtb, vaddr, info);
     } else {
         errprint("Invalid paging mode during vmi_pagetable_lookup\n");
     }
 
-    if(info->paddr) {
-        ret = VMI_SUCCESS;
+    /* add this to the cache */
+    if (ret == VMI_SUCCESS) {
+        v2p_cache_set(vmi, vaddr, dtb, info->paddr);
+    }
+    return ret;
+}
+
+/*
+ * Return a status when page_info is not needed, but also use the cache,
+ * which vmi_pagetable_lookup_extended() does not do.
+ *
+ * TODO: Should this eventually replace vmi_pagetable_lookup() in the API?
+ */
+status_t vmi_pagetable_lookup_cache(
+    vmi_instance_t vmi,
+    addr_t dtb,
+    addr_t vaddr,
+    addr_t *paddr)
+{
+    status_t ret = VMI_FAILURE;
+    page_info_t info = {0};
```

Comment:
I think it's better to initialize the struct here with {.dtb=dtb, .vaddr=vaddr}. The memset to 0 is really not required (and btw you already did that once when initialized the struct with {0}).